gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java

```diff
@@ -1390,6 +1445,40 @@ protected void configureRestModelGeneration(Project project, SourceSet sourceSet
     });
   }
 
+  protected void configurePegasusSchemaSnapshotGeneration(Project project, SourceSet sourceSet, boolean isExtensionSchema)
+  {
+    File schemaDir = isExtensionSchema? project.file(getExtensionSchemaPath(project, sourceSet))
+        : project.file(getDataSchemaPath(project, sourceSet));
+    if (isExtensionSchema && !SharedFileUtils.getSuffixedFiles(project, schemaDir, PDL_FILE_SUFFIX).isEmpty())
+    {
+      project.getLogger().debug("No extension schemas, skip generatePegasusExtensionSchemaSnapshot task");
+      return;
+    }
+    File publishablePegasusSchemaSnapshotDir = project.file(project.getBuildDir().getAbsolutePath()
+        + File.separatorChar + sourceSet.getName() + (isExtensionSchema ? PEGASUS_EXTENSION_SCHEMA_SNAPSHOT: PEGASUS_SCHEMA_SNAPSHOT));
```

Comment:
It shouldn't be necessary to build a String and pass it through `Project#file`. I would recommend using `java.nio.file.Path` whenever possible, since it allows you to write:

```
project.getBuildDir().toPath().resolve(sourceSet.getName() + ...)
```

Gradle task properties can be `Path` instead of `File` as well, and it is also possible to convert from a `Path` to a `File` using `Path#toFile`.

Reply:
Updated based on suggestion